kitsune/messages/utils.py

```diff
@@ -18,22 +19,37 @@ def send_message(to, to_group=None, text=None, sender=None):
 
     # Assuming OutboxMessage doesn't necessarily need to be created for every recipient
     if to:
+        # This is users from the To field
+        users = {user for user in to if isinstance(user, User)}
+        # This is groups from the To field
+        groups = [group for group in to if isinstance(group, Group)]
+        # Lets get all the data for groups that we need
+        prefetched_groups = Group.objects.filter(
+            pk__in=[group.pk for group in groups]
+        ).prefetch_related("user_set")
+        # This is who is actually getting a message
+        # - A combo of users from To field and users from groups
+        receivers = users | {user for group in prefetched_groups for user in group.user_set.all()}
+
```

Comment:
I think directly getting the users might be a better way to go? In other words, something like:

```python
        receivers = users | set(User.objects.filter(groups__in=groups).distinct())
```

Reply:
I'll try this - I thought `.distinct()` would be an expensive choice, but maybe this combination makes that a wash.

Reply:
Good point. I was debating that as well, whether we should drop the `.distinct()` and just let the `set()` take care of it. I was thinking that the `distinct` call in Postgres might be faster, but I really don't know.